WebHdfs.Test/Client.cs

```diff
@@ -6,189 +6,191 @@
-        public void SetReplicationFactor()
-        {
-            var path = "/path/to/file";
-            var param = 100;
-            CallClient(c => c.SetReplicationFactor(path, param).Wait(), HttpMethod.Put, path, "SETREPLICATION&replication=" + param, BOOL_RESULT);
-        }
-
-        [Test]
-        public void GetEmptyResult()
-        {
-            var path = "/path/to/file";
-            CallClient(async c =>
-                {
-                    var file = await c.GetFileStatus(path);
-                    Assert.IsNull(file);
-                }, HttpMethod.Get, path, "GETFILESTATUS", status: HttpStatusCode.NotFound);
-        }
-
-        private void CallClient(Action<WebHdfsClient> caller, HttpMethod method, string url, string operation, string result = "{}", HttpStatusCode status = HttpStatusCode.OK)
-		{
-			var handler = new Mock<FakeHttpMessageHandler>();
-			handler.CallBase = true;
-
-            Expression<Func<FakeHttpMessageHandler, HttpResponseMessage>> homeCall = t => t.Send(It.Is<HttpRequestMessage>(
-                            msg =>
-                               msg.Method == HttpMethod.Get &&
-                               msg.RequestUri.ToString() == "http://test.me/plz/webhdfs/v1/?user.name=hdfs&op=GETHOMEDIRECTORY"));
-
-            handler.Setup(homeCall)
-                   .Returns(new HttpResponseMessage(HttpStatusCode.OK) { Content = new StringContent("{\"Path\":\"/user/hdfs\"}", System.Text.Encoding.UTF8, "application/json") })
-                   .Verifiable();
-
-
-            if (!operation.StartsWith("GETHOMEDIRECTORY", StringComparison.OrdinalIgnoreCase))
-            {
-                Expression<Func<FakeHttpMessageHandler, HttpResponseMessage>> innerCall = t => t.Send(It.Is<HttpRequestMessage>(
-                    msg =>
-                        msg.Method == method &&
-                        msg.RequestUri.ToString().StartsWith(BASE_URL + WebHdfsClient.PREFIX + url + "?user.name=" + USER + "&op=" + operation, StringComparison.OrdinalIgnoreCase)));
-
-                handler.Setup(innerCall)
-                        .Returns(new HttpResponseMessage(status) { Content = new StringContent(result, System.Text.Encoding.UTF8, "application/json") })
-                        .Verifiable();
-            }
-
-            var client = new WebHdfsClient(handler.Object, BASE_URL, USER);
-			caller(client);
+		private const string TEST_FILE = "test.txt";
+		private const string BOOL_RESULT = "{ \"boolean\" : true }";
+
+		[Fact]
+		public async Task GetStatus()
+		{
+			const string path = "/path/to/file";
+			await CallClient(c => c.GetFileStatus(path), HttpMethod.Get, path, "GETFILESTATUS");
+		}
+
+		[Fact]
+		public async Task CreateDirectory()
+		{
+			const string path = "/path/to/file";
+			await CallClient(c => c.CreateDirectory(path), HttpMethod.Put, path, "MKDIRS");
+		}
+		
+		[Fact]
+		public async Task CreateFile()
+		{
+			if (!File.Exists(TEST_FILE))
+				using (File.Create(TEST_FILE)){ }
```

Comment:
Forgot to mention this. Not sure if it's right? The tests were failing before and after my changes. I assume you had a test.txt locally that wasn't included in the repo?